server/src/main/java/com/objectcomputing/checkins/services/file/FileServicesImpl.java

```diff
@@ -0,0 +1,226 @@
+    private final EmailSender emailSender;
+    private final SecurityService securityService;
+    private final CheckInServices checkInServices;
+    private final CheckinDocumentServices checkinDocumentServices;
+    private final MemberProfileServices memberProfileServices;
+
+    public FileServicesImpl(GoogleDriveAccessor googleDriveAccessor, EmailSender emailSender,
+                            SecurityService securityService, CheckInServices checkInServices,
+                            CheckinDocumentServices checkinDocumentServices,
+                            MemberProfileServices memberProfileServices) {
+        this.googleDriveAccessor = googleDriveAccessor;
+        this.emailSender = emailSender;
+        this.securityService = securityService;
+        this.checkInServices = checkInServices;
+        this.checkinDocumentServices = checkinDocumentServices;
+        this.memberProfileServices = memberProfileServices;
+    }
+
+    @Override
+    public HttpResponse<?> findFiles(@Nullable UUID checkInID) {
+
+        Boolean isAdmin = securityService!=null ? securityService.hasRole(RoleType.Constants.ADMIN_ROLE) : false;
+        validate(checkInID == null && !isAdmin, "You are not authorized to perform this operation");
+
+        try {
+            Set<FileInfoDTO> result = new HashSet<>();
+            Drive drive = googleDriveAccessor.accessGoogleDrive();
+            validate(drive == null, "Unable to access Google Drive");
+
+            if (checkInID == null && isAdmin) {
+                //find all
+                FileList fileList = drive.files().list().execute();
+                for (File file : fileList.getFiles()) {
+                    result.add(setFileInfo(file, null));
+                }
+            } else if (checkInID != null) {
+                //find by CheckIn ID
+                CheckIn checkIn = checkInServices.read(checkInID);
+                validate(checkIn == null, "Unable to find checkin record with id %s", checkInID);
+
+                Set<CheckinDocument> checkinDocuments = checkinDocumentServices.read(checkInID);
+                for (CheckinDocument cd : checkinDocuments) {
+                    File file = drive.files().get(cd.getUploadDocId()).execute();
+                    result.add(setFileInfo(file, cd));
+                }
+            }
+
+            return HttpResponse.ok(result);
+        } catch (GoogleJsonResponseException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse
+                    .status(HttpStatus.valueOf(e.getStatusCode()))
+                    .body(e.getContent());
+        } catch (IOException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public HttpResponse<?> downloadFiles(@NotNull String uploadDocId) {
```

Comment:
It's missing from the DoD, but this endpoint should follow the same validation as findFiles.

Reply:
updated in the latest commit.